cpu/stm32f4/include/periph_cpu.h

```diff
@@ -115,6 +140,13 @@ typedef struct {
 void gpio_init_af(gpio_t pin, gpio_af_t af);
 
 /**
+ * @brief   Configure the given pin to be used as ADC input
+ *
+ * @param[in] pin       pin to configure
+ */
+void gpio_init_analog(gpio_t pin);
```

Comment:
I was thinking about it already in an other situation, but don't you think it's possible to define these functions as public prototypes for peripheral drivers?